cfgov/agreements/metadata.py

```diff
@@ -0,0 +1,29 @@
+from django.conf import settings
+
+import boto3
+
+
+class Metadata:
+    def __init__(self):
+        s3 = boto3.resource("s3")
+        self.bucket = s3.Bucket(settings.AWS_STORAGE_BUCKET_NAME or "")
+        self.prefix = "a/assets/bulk_agreements/"
+        self.flexibilities = [
+            "Q1-2020",
+            "Q2-2020",
+            "Q3-2020",
+            "Q4-2020",
+            "Q1-2021",
+        ]
+        self.notes = {
+            "Q2-2019": "Agreements are incomplete due to technical submission"
+            " issues at the Bureau"
+        }
+
+    def get_objects_by_prefix(self):
+        return self.bucket.objects.filter(Prefix=self.prefix)
+
+    def get_sorted_agreements(self, reverse=True):
+        agreements = [f.key for f in self.get_objects_by_prefix()]
+        agreements.sort(reverse=reverse)
+        return agreements
```

Comment:
[Given the list of possible boto errors](https://github.com/boto/botocore/blob/develop/botocore/exceptions.py), it seems like `ConnectionError` is the one that's most likely not our fault (e.g. S3 is down or something). What do you think about this?:

```suggestion
from botocore.exceptions import ConnectionError

# ...

    def get_sorted_agreements(self, reverse=True):
        try:
            agreements = [f.key for f in self.get_objects_by_prefix()]
            agreements.sort(reverse=reverse)
            return agreements
        except ConnectionError:
            return []
```

Reply:
I haven't read all of the error-handling documentation yet, it might be helpful too: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html

Reply:
Cooool, added this along with `ClientError` which is what is thrown if credentials are bad.